src/features/artist/pages/Artist.tsx

```diff
@@ -1,41 +1,48 @@
 import { useRoute } from '@react-navigation/native'
-import React, { FunctionComponent } from 'react'
+import React, { FunctionComponent, useMemo } from 'react'
 
 import { ArtistBody } from 'features/artist/components/ArtistBody/ArtistBody'
+import { findArtistById } from 'features/artist/helpers/findArtistById'
 import { PageNotFound } from 'features/navigation/pages/PageNotFound'
 import { UseRouteType } from 'features/navigation/RootNavigator/types'
-import { getOfferArtists } from 'features/offer/helpers/getOfferArtists/getOfferArtists'
+import { useArtistResults } from 'features/offer/helpers/useArtistResults/useArtistResults'
 import { useFeatureFlag } from 'libs/firebase/firestore/featureFlags/useFeatureFlag'
 import { RemoteStoreFeatureFlags } from 'libs/firebase/firestore/types'
-import { useSubcategoriesMapping } from 'libs/subcategories'
-import { useOfferQuery } from 'queries/offer/useOfferQuery'
 
 export type Artist = {
+  id: string
   name: string
   bio?: string
+  image?: string
 }
 
 export const Artist: FunctionComponent = () => {
   const enableArtistPage = useFeatureFlag(RemoteStoreFeatureFlags.WIP_ARTIST_PAGE)
   const { params } = useRoute<UseRouteType<'Artist'>>()
-  const { data: offer } = useOfferQuery({ offerId: params.fromOfferId })
-  const subcategoriesMapping = useSubcategoriesMapping()
 
-  if (!offer) return null
+  const { artistPlaylist, artistTopOffers } = useArtistResults({
+    artistId: params.id,
+  })
 
-  // TODO(PC-33464): point of vigilance when we will use a hook which get the artists from Algolia
-  const subcategory = subcategoriesMapping[offer?.subcategoryId]
-  const artists = getOfferArtists(subcategory.categoryId, offer)
+  const artist = useMemo(() => {
+    return findArtistById(artistPlaylist, params.id) || findArtistById(artistTopOffers, params.id)
```

Comment:
Ici au lieu d'appeler ta méthode 2 fois ne pourrais-tu pas faire quelque chose comme ça ? : 
```suggestion
    return findArtistById([...artistPlaylist,...artistTopOffers], params.id)
```

Reply:
En fait tu pourrais même te baser sur une offre au pif normalement non ?
Elles sont censées toutes avoir l'artiste que tu cherches.

Et (encore mieux selon moi), normalement à chaque fois que t'arrives sur la page Artiste, t'es censé avoir déjà les données `id`, `name` et `image`, on pourrait pas les passer en paramètre au screen ?
Ça éviterait d'attendre la requête Algolia pour ensuite tout parser et enfin afficher des données qu'on avait en fait déjà.
Je sais pas si c'est facile

Reply:
La 1ère option pourquoi pas, @mmeissonnier-pass t'en penses quoi ?

La 2ème j'en ai parlé avec @yleclercq-pass et on n'était pas forcément chaud pour mettre toutes les infos dans les params de l'URL. @mmeissonnier-pass un avis ?

Reply:
d'accord pour l'option 1.

Pour l'option 2, plutôt que d'utiliser des params dans l'URL (pas trop fan), on pourrait récupérer le name et l'image directement depuis le `queryClient` avec la bonne queryKey basée sur l'id de l'artiste ?

Reply:
@mmeissonnier-pass On n'a pas d'API pour requêter l'artiste, on le récupère via les offres donc si j'ai bien compris ton option 2 c'est pas possible en l'état

Reply:
Ah ! Mais j'ai besoin de l'image aussi ! Après selon moi ça reste quand même un appel inutile car on a l'info dans les offres et dans tous les cas il faut charger les 2 playlist